cdap-common/src/main/java/co/cask/cdap/common/namespace/RemoteNamespaceQueryClient.java

```diff
@@ -42,26 +48,41 @@
  */
 public class RemoteNamespaceQueryClient extends AbstractNamespaceQueryClient {
   private final Supplier<EndpointStrategy> endpointStrategySupplier;
+  private final String userId;
 
   @Inject
-  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient) {
+  RemoteNamespaceQueryClient(final DiscoveryServiceClient discoveryClient, CConfiguration cConf) {
     this.endpointStrategySupplier = Suppliers.memoize(new Supplier<EndpointStrategy>() {
       @Override
       public EndpointStrategy get() {
-        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.REMOTE_SYSTEM_OPERATION));
+        return new RandomEndpointStrategy(discoveryClient.discover(Constants.Service.APP_FABRIC_HTTP));
       }
     });
+    try {
+      if (SecurityUtil.isKerberosEnabled(cConf)) {
```

Comment:
This does not seem to work for impersonated namespace. The issue is that we  cannot always use cdap master keytab since cdap user will have no permission on an impersonated namespace being created by the user. The namespace meta for the namespace will belong to the user and any get operation on this namespace will fail. 
I think we should pass the AuthenticationContext.getPrincipal().getName() if kerberos is enabled.
@bdmogal : Any suggestions on how to solve this ?